common/src/main/java/org/apache/celeborn/common/network/server/MapDataPartition.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.network.server;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.celeborn.common.meta.FileInfo;
+import org.apache.celeborn.common.network.server.memory.BufferQueue;
+import org.apache.celeborn.common.network.server.memory.BufferRecycler;
+import org.apache.celeborn.common.network.server.memory.MemoryManager;
+import org.apache.celeborn.common.network.server.memory.ReadBufferRequest;
+import org.apache.celeborn.common.util.JavaUtils;
+
+// this means active data partition
+class MapDataPartition implements MemoryManager.ReadBufferTargetChangeListener {
+  public static final Logger logger = LoggerFactory.getLogger(MapDataPartition.class);
+  private final List<Long> activeStreamIds = new ArrayList<>();
+  private final FileInfo fileInfo;
+  private final ExecutorService readExecutor;
+  private final ConcurrentHashMap<Long, MapDataPartitionReader> readers =
+      JavaUtils.newConcurrentHashMap();
+  private FileChannel dataFileChanel;
+  private FileChannel indexChannel;
+  private volatile boolean isReleased = false;
+  private final BufferQueue bufferQueue = new BufferQueue();
+  private boolean bufferQueueInitialized = false;
+  private MemoryManager memoryManager = MemoryManager.instance();
+  private Consumer<Long> recycleStream;
+  private volatile long localMemoryTarget = 0;
+  private volatile int localBuffersTarget = 0;
```

Comment:
Should move this variable in BufferQueue, and let BufferQueue to ensure consistency.

Reply:
NO.